server/log_rpc_server_test.go

```diff
@@ -1146,178 +1146,167 @@ func TestGetConsistencyProofBeginTXFails(t *testing.T) {
+	nodeIDs     []storage.NodeID
+	hashes      [][]byte
+	nodes       []storage.Node
+	wantErr     bool
+	noSnap      bool
+	noRev       bool
+	noRoot      bool
+	noClose     bool
+	noCommit    bool
+	snapErr     error
+	rootErr     error
+	getNodesErr error
+	commitErr   error
 }
 
-func TestGetConsistencyProofCommitFails(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-
-	test := newParameterizedTest(ctrl, "GetConsistencyProof", readOnly,
-		func(t *storage.MockLogTreeTX) {
-			t.EXPECT().LatestSignedLogRoot(gomock.Any()).Return(signedRoot1, nil)
-			t.EXPECT().ReadRevision().Return(signedRoot1.TreeRevision)
-			t.EXPECT().GetMerkleNodes(gomock.Any(), revision1, nodeIdsConsistencySize4ToSize7).Return([]storage.Node{{NodeID: stestonly.MustCreateNodeIDForTreeCoords(2, 1, 64), NodeRevision: 3}}, nil)
+func TestGetConsistencyProof(t *testing.T) {
+	tests := []consistProofTest{
+		{
+			// Storage snapshot fails, should result in an error. Happens before we have a TX so
+			// no Close() etc.
+			req:      getConsistencyProofRequest7,
+			snapErr:  errors.New("SnapshotForTree() failed"),
+			wantErr:  true,
+			noRev:    true,
+			noRoot:   true,
+			noCommit: true,
+			noClose:  true,
+		},
+		{
+			// Storage fails to read the log root, should result in an error.
+			req:      getConsistencyProofRequest7,
+			rootErr:  errors.New("LatestSignedLogRoot() failed"),
+			wantErr:  true,
+			noRev:    true,
+			noCommit: true,
+		},
+		{
+			// Storage fails to get nodes, should result in an error
+			req:         getConsistencyProofRequest7,
+			nodeIDs:     nodeIdsConsistencySize4ToSize7,
+			hashes:      [][]byte{[]byte("nodehash")},
+			nodes:       []storage.Node{{NodeID: stestonly.MustCreateNodeIDForTreeCoords(2, 1, 64), NodeRevision: 3, Hash: []byte("nodehash")}},
+			wantErr:     true,
+			getNodesErr: errors.New("getMerkleNodes() failed"),
+			noCommit:    true,
+		},
+		{
+			// Storage fails to commit, should result in an error.
+			req:       getConsistencyProofRequest7,
+			nodeIDs:   nodeIdsConsistencySize4ToSize7,
+			hashes:    [][]byte{[]byte("nodehash")},
+			nodes:     []storage.Node{{NodeID: stestonly.MustCreateNodeIDForTreeCoords(2, 1, 64), NodeRevision: 3, Hash: []byte("nodehash")}},
+			wantErr:   true,
+			commitErr: errors.New("Commit() failed"),
+		},
+		{
+			// Storage doesn't return the requested node, should result in an error.
+			req:      getConsistencyProofRequest7,
+			nodeIDs:  nodeIdsConsistencySize4ToSize7,
+			hashes:   [][]byte{[]byte("nodehash")},
+			nodes:    []storage.Node{{NodeID: stestonly.MustCreateNodeIDForTreeCoords(3, 1, 64), NodeRevision: 3, Hash: []byte("nodehash")}},
+			wantErr:  true,
+			noCommit: true,
+		},
+		{
+			// Storage returns an unexpected extra node, should result in an error.
+			req:     getConsistencyProofRequest7,
+			nodeIDs: nodeIdsConsistencySize4ToSize7,
+			hashes:  [][]byte{[]byte("nodehash")},
+			nodes: []storage.Node{
```

Comment:
(nit) indentation looks hinky here

Reply:
It seems to confuse the formatter. It works OK if nodes is all on one line but I was trying to make it read easier.